plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/command/AssociationDeleteCommand.java

```diff
@@ -65,21 +72,20 @@ public AssociationDeleteCommand(AssociationPart part) {
     public void execute() {
         DBSEntityAssociation entityAssociation = association.getObject();
         if (entityAssociation instanceof DBVEntityForeignKey) {
-            if (!UIUtils.confirmAction("Delete logical key", "Are you sure you want to delete logical key '" + part.getAssociation().getName() + "'?")) {
+            if (!UIUtils.confirmAction("Delete logical key", "Are you sure you want to delete logical key '" + association.getName() + "'?")) {
                 return;
             }
-            DBVEntity vEntity = DBVUtils.getVirtualEntity(entityAssociation.getParentObject(), false);
             if (vEntity == null) {
                 UIUtils.showMessageBox(UIUtils.getActiveWorkbenchShell(), "No virtual entity", "Can't find association owner virtual entity", SWT.ICON_ERROR);
                 return;
             }
             vEntity.removeForeignKey((DBVEntityForeignKey) entityAssociation);
+            vEntity.persistConfiguration();
```

Comment:
Configuration is already persisting in `ERDEditorEmbedded::doSave`. Not sure if it should be done here too.